### PR TITLE
update forums, docs, snack, blog, website, and email links to their expo.dev equivalents

### PR DESCRIPTION
**Pull request by FiberJW**
# Why

Our new public domain is now expo.dev

# How

- Searched for forums, blog, docs, snack, website and emails and replaced them with their new domains. 
- Changed expo.io/--/ API links to exp.host

### CODE REVIEW

**Comment by FiberJW** [comment]
That's a lot of reviewers 👀

File: ios/Tests/Linking/EXLinkingTests.m [hunk 1]
```diff
@@ -70,10 +70,10 @@ - (void)testDoesDeepLinkingRespectCustomReleaseChannel
 
 - (void)testDoesDeepLinkingDifferentiateDomains
 {
-  [self _assertDeepLink:@"https://expo.io/@ben/foodwheel" doesNotRouteToManifest:@"https://exp.host/@ben/foodwheel"];
+  [self _assertDeepLink:@"https://expo.dev/@ben/foodwheel" doesNotRouteToManifest:@"https://exp.host/@ben/foodwheel"];
   [self _assertDeepLink:@"https://google.com/@ben/foodwheel" doesNotRouteToManifest:@"https://exp.host/@ben/foodwheel"];
   [self _assertDeepLink:@"https://@ben/foodwheel" doesNotRouteToManifest:@"https://exp.host/@ben/foodwheel"];
-  [self _assertDeepLink:@"https://expo.io/@ben/foodwheel" doesNotRouteToManifest:@"https://@ben/foodwheel"];
+  [self _assertDeepLink:@"https://expo.dev/@ben/foodwheel" doesNotRouteToManifest:@"https://@ben/foodwheel"];
```
Comment (ide) [inline ios/Tests/Linking/EXLinkingTests.m:76]: revert and rethink this file

File: ios/Tests/AppLoader/EXFileDownloaderTests.m [hunk 2]
```diff
@@ -47,7 +47,7 @@ - (void)testDoesDefaultFileDownloaderDownloadSomething
 {
   XCTestExpectation *expectToDownload = [[XCTestExpectation alloc] initWithDescription:@"Default EXFileDownloader should download a json file"];
   EXFileDownloader *fileDownloader = [[EXFileDownloader alloc] init];
-  NSURL *jsonFileUrl = [NSURL URLWithString:@"https://expo.io/@exponent/home/index.exp"];
+  NSURL *jsonFileUrl = [NSURL URLWithString:@"https://expo.dev/@exponent/home/index.exp"];
```
Comment (ide) [inline ios/Tests/AppLoader/EXFileDownloaderTests.m:50]:
```suggestion
  NSURL *jsonFileUrl = [NSURL URLWithString:@"https://exp.host/@exponent/home/index.exp"];
```

File: guides/releasing/Versions Endpoint.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Updating the Versions endpoint
 
-The [versions endpoint](https://expo.io/--/api/v2/versions) ([staging](https://staging.expo.io/--/api/v2/versions)) returns a JSON object containing information that is useful primarily in expo-cli for installing simulator builds, validating package versions, ejecting, and upgrading packages. It can also be used for any other purposes where similar information is required, for example linking to release notes from the website.
+The [versions endpoint](https://exp.host/--/api/v2/versions) ([staging](https://staging.exp.host/--/api/v2/versions)) returns a JSON object containing information that is useful primarily in expo-cli for installing simulator builds, validating package versions, ejecting, and upgrading packages. It can also be used for any other purposes where similar information is required, for example linking to release notes from the website.
```
Comment (ide) [inline guides/releasing/Versions Endpoint.md:3]: :+1:

**Review by ide — CHANGES_REQUESTED** [review]
Revert docs/scripts/schemas/ since they are synced from www.

**Pull request closed without merging**